### PR TITLE
[opt](compaction) Don't check missed rows in cumu compaction if input rowsets are not in tablet

### PR DESCRIPTION
**Pull request by bobhan1**
### What problem does this PR solve?

Problem Summary:

Suppose a heavy schema change process on BE converting tablet A to tablet B.
1. during schema change double write, new loads write [X-Y] on tablet B.
2. rowsets with version [a],[a+1],...,[b-1],[b] on tablet B are picked for cumu compaction(X<=a<b<=Y).(cumu compaction on new tablet during schema change double write is allowed after https://github.com/apache/doris/pull/16470)
3. schema change remove all rowsets on tablet B before version Z(b<=Z<=Y) before it begins to convert historical rowsets.
4. schema change finishes.
5. cumu compation begins on new tablet with version [a],...,[b]. If there are duplicate keys between these rowsets, the compaction check will fail because these rowsets have skipped to calculate delete bitmap in commit phase and publish phase because tablet B is in NOT_READY state when writing. 

This PR makes cumu compaction skip to do missed rows check when it finds out that the input rowsets are not exist in tablet.

Note that the compaction will fail because `Tablet::modify_rowsets` will check if rowsets in `to_delete` still exist in tablet's `_rs_version_map`.

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [x] Confirm the release note
- [x] Confirm test cases
- [x] Confirm document
- [x] Add branch pick label

### CODE REVIEW

File: be/src/olap/compaction.cpp [hunk 1]
```diff
@@ -986,8 +986,33 @@ Status CompactionMixin::modify_rowsets() {
...
+            bool need_to_check_missed_rows = true;
+            {
+                std::shared_lock rlock(_tablet->get_header_lock());
+                need_to_check_missed_rows =
+                        std::all_of(_input_rowsets.begin(), _input_rowsets.end(),
+                                    [&](const RowsetSharedPtr& rowset) {
+                                        return tablet()->rowset_exists_unlocked(rowset);
+                                    });
+            }
+            if (!need_to_check_missed_rows) {
```
Comment (zhannngchen) [inline be/src/olap/compaction.cpp:999]: don't need such log here, the compaction will fail finally.

**Comment by bobhan1** [comment]
run buildall

File: be/src/olap/compaction.cpp [hunk 2]
```diff
@@ -986,8 +986,19 @@ Status CompactionMixin::modify_rowsets() {
             if (!_tablet->tablet_meta()->tablet_schema()->cluster_key_uids().empty()) {
                 merged_missed_rows_size += _stats.filtered_rows;
             }
+
+            bool need_to_check_missed_rows = true;
```
Comment (zhannngchen) [inline be/src/olap/compaction.cpp:990]: please add comment here.

**Comment by bobhan1** [comment]
run buildall

**Review by zhannngchen — APPROVED** [review]
LGTM

**Review by hust-hhb — APPROVED** [review]
LGTM

**Pull request merged**